### ISSUE TITLE
Google Summer of Code 2021 planning

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#10072]
Mypy will apply to participate in Google Summer of Code 2021. **The deadline for applications is Feb 19 so we need to move quickly.**

I'm planning to mentor one student and act as our org admin.

Each student works on a project for 10 weeks, an average of 18 hours a week (roughly the same as 6 30-hour weeks). For more information, see the [GSoC web site](https://summerofcode.withgoogle.com/).

We can also have additional mentors. If you think that you have the time to spare and enough experience working on mypy (or mypyc) to mentor a student, feel free to contact me (or leave a comment here). You can propose a project you'd like to mentor, or we can figure out a suitable project together.

My project proposal is going to be focused on mypyc. The goal is to speed up string operations. It would involve adding new primitives for `str` and `bytes` operations, and a faster implementation of string formatting (f-strings, `str.format` and/or `%` formatting). Additional tasks include designing or finding realistic benchmarks for string operations, analyzing open source codebases to determine which string operations are common, and validating that any optimizations result in improved performance.

Next steps:
- [x] Create an issue that describes the project I'm proposing in more detail (#10099)
- [x] Submit the GSoC org application (by Feb 19)
- [x] Prepare starter issues for students who are interested in applying (we'll probably expect some prior contributions before we can accept a student) (#10195)
- [x] Wait for decision from GSoC (by Mar 9) (Updates: mypy has been selected as a GSoC org this year!)

**Comment by PrasanthChettri** [comment]
@JukkaL I have little experience in the org, I have worked on one issue #10053 and I would like to be contribute to your proposal, could you point me to some starter issues that are more beginner friendly revolving around mypyc ?

**Comment by Avirup101** [comment]
@JukkaL I would love to work on the project under your mentorship. Would you please like to tell me about some starter issues which can be fixed as a beginner to start off with?

**Comment by 97littleleaf11** [comment]
Hi, @JukkaL . I am interested in contributing to mypyc project. I just made a PR #10088 related to `str` primitives as a small beginning.  It would be a great honor to work under your supervision.

**Comment by JukkaL** [comment]
@97littleleaf11 Sounds great! We are currently busy preparing to submit our org application, but I'll be happy to help once we get the administrative work done.

I added a separate issue about all project ideas (currently only the one I mentioned above, but we may add more): #10099. There's also a link to a more detailed description of the project.

**Comment by JukkaL** [comment]
I've submitted the GSoC org application.

**Comment by hauntsaninja** [comment]
In case anyone here is interested, here are some high impact "good second issues": https://github.com/python/mypy/issues/9880 https://github.com/python/mypy/issues/8481#issuecomment-739446125 https://github.com/python/mypy/issues/9852

**Comment by JukkaL** [comment]
I created an issue with ideas about good first and second issues: #10195 

Please add new ideas about issues and questions about contributing there.

**Comment by ethanhs** [comment]
It looks like mypy is not a GSoC org this year, as we don't seem to be listed. Also I'm closing this since we are no longer in 2021 :)
